File: 509-fibonacci-number/509-fibonacci-number.cpp
class Solution {
public:
    int fib(int n) {
        if (n==0)
            return 0;
        else if( n==1)
            return 1;
        else{
            int x=0;
            int y=1;
            int sm=0;
            for(int i = 2 ; i<n+1;i++){
                sm=x+y;
                x=y;
                y=sm;
                }
            return sm;
        }
    }
};

File: 509-fibonacci-number/509-fibonacci-number.py
class Solution:
    def fib(self, n: int) -> int:
        
        if n==0:
            return 0
        elif n==1:
            return 1
        else:
            x=0
            y=1
            sm=0
            for _ in range(2,n+1):
                sm=x+y
                x=y
                y=sm
            return sm
                